src/modules/billing/controllers/invoices.js

```diff
@@ -1,27 +1,14 @@
-const { BATCH_STATUS } = require('../../../lib/models/batch');
-const Boom = require('@hapi/boom');
+'use strict';
+
 const invoiceService = require('../../../lib/services/invoice-service');
 const mapErrorResponse = require('../../../lib/map-error-response');
 
-const invoiceIsPartOfSentBatch = invoice => {
-  const { billingBatch } = invoice;
-  return billingBatch.status === BATCH_STATUS.sent;
-};
-
-const invoiceIsRebill = invoice =>
-  invoice.rebillingState !== null;
-
 const patchInvoice = async request => {
   try {
     const { invoiceId } = request.params;
-    const invoice = await invoiceService.getInvoiceById(invoiceId);
-    if (!invoiceIsPartOfSentBatch(invoice)) {
-      return Boom.conflict('Cannot update invoice that is not part of a sent batch');
-    }
-    if (invoiceIsRebill(invoice)) {
-      return Boom.conflict('Cannot update invoice that is itself a rebill');
-    }
-    return invoiceService.updateInvoice(invoiceId, request.payload);
+    const { isFlaggedForRebilling } = request.payload;
+
+    return await invoiceService.setIsFlaggedForRebilling(invoiceId, isFlaggedForRebilling);
```

Comment:
Is `return await...` valid syntax?

I would expect `return [the Promise]` normally

Reply:
Yes it is, it means any error that occurs in the promise will be handled by the try/catch, whereas this is not the case if you return the promise directly